### PR TITLE
Add download update instructions "how to release" documentation

### PR DESCRIPTION
**Pull request by TcT2k**
Updated instructions will be relevant after merge of [website PR](https://github.com/wxWidgets/website/pull/27) to improve the downloads page

### CODE REVIEW

File: docs/contributing/how-to-release.md [hunk 1]
```diff
@@ -117,8 +117,12 @@ Create https://docs.wxwidgets.org/x.y.z/ (ask Bryan to do it if not done yet).
 
 ## Announcement
 
-Update https://www.wxwidgets.org, usually a news item is enough but something
-more can be called for for major releases.
+Update https://www.wxwidgets.org:
+* Update release information (at least `version` and `released`) in `_data/relases.yml`.
+* Download information can than be updated by running `update_release_info.rb`.
```
Comment (PBfordev) [inline docs/contributing/how-to-release.md]: I think _than_ should be changed to _th**e**n_

File: docs/contributing/how-to-release.md [hunk 2]
```diff
@@ -117,8 +117,12 @@ Create https://docs.wxwidgets.org/x.y.z/ (ask Bryan to do it if not done yet).
...
+  This will update the asset information from github.
+* Add a news item. Usually a news item is enough but something
+  more can be called for for major releases
 
 Post `docs/publicity/announce.txt` at least to wx-announce@googlegroups.com and
 to wx-users for the important releases.
```
Comment (PBfordev) [inline docs/contributing/how-to-release.md]: Considering the frequency of releases, I think all releases in both branches could be announced in wx-users.

File: docs/contributing/how-to-release.md [hunk 3]
```diff
@@ -117,8 +117,12 @@ Create https://docs.wxwidgets.org/x.y.z/ (ask Bryan to do it if not done yet).
 
 ## Announcement
 
-Update https://www.wxwidgets.org, usually a news item is enough but something
-more can be called for for major releases.
+Update https://www.wxwidgets.org:
+* Update release information (at least `version` and `released`) in `_data/relases.yml`.
+* Download information can than be updated by running `update_release_info.rb`.
+  This will update the asset information from github.
```
Comment (PBfordev) [inline docs/contributing/how-to-release.md]: GitHub is usually spelled in wxWidgets docs as **G**it**H**ub except for when it is a part of an URL.

**Comment by TcT2k** [comment]
@PBfordev thanks for looking, I've fixed/included your comments.

**Pull request merged**